bftengine/src/bftengine/messages/ViewChangeMsg.cpp

```diff
@@ -178,6 +216,32 @@ bool ViewChangeMsg::checkElements(uint16_t sigSize) const {
   return true;
 }
 
+bool ViewChangeMsg::checkComplaints(uint16_t sigSize) const {
+  uint16_t numOfActualComplaints = 0;
+  uint32_t remainingBytes = size() - sigSize - b()->locationAfterLast;
+  char* currLoc = body() + b()->locationAfterLast + sigSize;
```

Comment:
I must admit, I got lost in the pointer arithmetic here...

Reply:
Yes, I have extracted the calculation of the body size in a separate function for better readability and to avoid mistakes. Also updated the visual representation :)